classes/Mail/MailMessage.php

```diff
@@ -101,20 +101,38 @@ private function ensureAttachment()
             }
             $this->attachments_cnt = 0;
             $parts = isset($structure->parts) ? $structure->parts : [ $structure ];
+
+            $allParts = [];
             foreach ($parts as $index => &$part) {
-                $att_part = $this->scanAttachmentPart($part, $index + 1);
+                $msgIndex = $index + 1;
+                // when it's an entire attached message: MESSAGE/RFC822
+                if (isset($part->parts) && count($part->parts) > 0) {
+                    foreach ($part->parts as $subIndex => &$subPart) {
+                        $allParts[$msgIndex . '.' . $subIndex + 1] = $subPart;
+                    }
+                    continue;
+                }
+                $allParts[$msgIndex] = $part;
+            }
+
```

Comment:
`unset($part);` is needed here either.

Reply:
Well, thanks, I will add it
in the current version it's out of the loop, and I find this weird
in my version I put in in the loop
But I am not sure if there is any reason to do this unset?

What comment coulf I add to this line of code?

Reply:
I missed that. It must be out of the loop too. See warning here: https://www.php.net/manual/en/control-structures.foreach.php

I use references here to avoid copying arrays in a loop.

Reply:
Wow, oookayy
Let me push a fix for this

Reply:
I'm not sure that there is a problem in this particular case, however, I always use unset after a reference is no longer needed.

Reply:
Do you want to pick the first commit or should I split it out?

Reply:
I'm going to apply the whole PR with all its commits at once. So no need to split them.